lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -584,7 +584,7 @@ public void forceDeleteTask(String taskId)
   }
 
   @Override
-  public Task selectAndClaim(TaskQuery taskQuery)
+  public Optional<Task> selectAndClaim(TaskQuery taskQuery)
```

Comment:
What do u think of this?
```
  @Override
  public Optional<Task> selectAndClaim(TaskQuery taskQuery) {
    ((TaskQueryImpl) taskQuery).selectAndClaimEquals(true);
    return taskanaEngine.executeInDatabaseConnection(
        () ->
            Optional.ofNullable(taskQuery.single()).map(TaskSummary::getId).map(wrap(this::claim)));
  }

```
This also removed the 2 thrown exceptions and thus their declaration can be removed aswell

Reply:
das funktioniert irgendwie nicht
lass uns gemeinsam drauf schauen

Reply:
fixed